packages/swarm-components/src/Textarea.jsx

```diff
@@ -13,25 +18,50 @@ type Props = {
     /**
      * Value for textarea.
      */
-    value?: string
+    value?: string,
+    /**
+     * max length of input field
+     */
+    maxLength?: string | number,
 }
 
 const getTextareaStatus = (props: Props): string => {
     return props.disabled ? 'disabled' : (props.error ? 'error' : 'default');
 };
 
 const getCharacterCount = (value: string = '') => value.length;
+type CharProps = {
+    children?: any
+}
+
+const CharCount = (props: CharProps) => {
+    return (
+        <p data-swarm-textarea-char-count className="text--tiny" {...props} />
+    );
+}
 
-const Textarea = (props: Props): React.Element<'textarea'> => {
+const Textarea = (props: Props): React.Element<'div'> => {
 
-    console.log('textarea char count', getCharacterCount(props.value));
+    const { maxLength, autosize } = props;
+
+    const textareaRef = React.useCallback(node => {
+        if (node !== null && props.autosize) {
+            auto(node);
+        }
+    }, [props.value]);
+
+    const remainingCharacters = (parseInt(maxLength, 10) - getCharacterCount(props.value))
 
     return (
-        <textarea
-            data-swarm-textarea={getTextareaStatus(props)}
-            {...props}
-        />
-    );
+        <div data-swarm-textarea-wrapper>
+            <textarea
+                data-swarm-textarea={getTextareaStatus({ ...props, error: props.error || remainingCharacters < 0 })}
```

Comment:
Sort of hard to read as a long inline expression. Maybe create a variable for this.

Reply:
Do you need to handle a state where `props.value` is undefined? It's optionally required in the flow type.  I'm assuming when there's no `props.value` we don't want an error state, but also want to show the maxLength as the character count?

Reply:
yes, that's the current behavior we want to keep.